docs/proposals/cli.md

```diff
@@ -0,0 +1,189 @@
+[more options](https://github.com/otaviof/shp/blob/55ce2e8d58435b0264e3db0bef5cf439abfeca18/vendor/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go#L317)
+out of the box.
+
+## Command
+
+The project will use [cobra](https://github.com/spf13/cobra)/[viper](https://github.com/spf13/viper)
+for scaffolding the command-line structure, which in terms of organization and workflow, follows
+`kubectl` convention. Thus, the following pattern:
+
+```sh
+$ shp <verb> <resource> <name> [options]
+```
+
+For instance:
+
+```sh
+$ shp create build nodejs-ex
+$ shp run build nodejs-ex
+```
+
+Alternatively, we may have a more conventional pattern, as in:
+
+```sh
+$ shp <resource> <verb> <name> [options]
+```
+
+For example:
+
+```sh
+$ shp create build build nodejs-ex
+$ shp run build nodejs-ex
+$ shp logs nodejs-ex
+```
+
+During the review process of this enhancement proposal, we must decide which pattern suits better,
+which one we prefer to go ahead with.
+
+## Features
+
+For the initial release, the aim is to be able to define a `Build`, run it and display logs.
+
+### Defining a Build
+
+```sh
+$ shp <verb> build <name> [options]
+```
+
+Will be responsible for managing Build objects, using the verbs: `create`, `update`, list and
+`delete`. Later on, end users should have the ability to bootstrap a local repository clone using a
+single command.
+
+### Running a Build
+
+```sh
+$ shp run build <name> [options]
+```
+
+This sub-command will instantiate a new BuildRun resource for a Build, effectively starting the
+build process. And, will print follow up commands that may be issued to see the BuildRun status
+and inspect logs.
+
+### Logs
+
+```sh
+$ shp logs <name> [options]
+```
+
+Retrieve all logs related to the informed `BuildRun` name. It will also be able to follow
+(`--follow`) container log output as they are executed. The log lines displayed are organised by
+the sequence of steps, and easy to read the whole build process output in a single go.
+
+Additionally, it must provide easy syntax to reach a specific `BuildRun` instance, or the most
+recent `BuildRun` for a `Build`.
+
+## More Features
+
+The command-line interface is planned to host more features, like for instance managing local and
+remove artifacts, as well as help end-users to upload data into the cluster.
+
+Thus, the initial design of the CLI must allow more subcommands to be added, accommodating more
+features which will depend in a client.
+
+## Testing
+
+The strategy to test the command-line application will be based on `go test` for the unit testing,
+and for end-to-end testing we should adopt [bats](https://github.com/sstephenson/bats). Bats
+framework gives us a structured way to run shell script commands and what we expect them to
+return, the command-line client shp is no more than a shell command.
+
+The testing structure will be composed of:
+
+- **Unit**: written in Golang and using [Gomega](https://onsi.github.io/gomega/) for assertion;
```

Comment:
Not sure if this belongs here, but I would prefer to forgo using Gomega and stick with pure go tests.